pkg/scm/gitlab/context.go

```diff
@@ -158,3 +164,47 @@ func (c *Context) HasExecutedActionGroup(group string) bool {
 
 	return ok
 }
+
+// GetCodeOwners returns the eligible code owners for the merge request
+//
+// This is based on the elibible approvers in the rules of the merge request approval
+// state api.
+func (c *Context) GetCodeOwners() []scm.Actor {
+	owners := []scm.Actor{}
+
+	for _, rule := range c.MergeRequest.ApprovalState.Rules {
+		// Multiple code owner paths could be matched when sections are used, so
+		// flatten the list of eligible approvers
+		if rule.Type.String() != "CODE_OWNER" {
```

Comment:
The `rule.Type` here is `nil`-able so we should guard against that, and then you can use the enums directly 

```go
if *rule.Type == ApprovalRuleTypeCodeOwner {
			continue
		}
```